src/iso15118/io/socket_helper.cpp

```diff
@@ -21,6 +22,7 @@ bool get_first_sockaddr_in6_for_interface(const std::string& interface_name, soc
     auto current_if = if_list_head;
 
     bool found_interface = false;
+    char buffer[INET6_ADDRSTRLEN];
 
     for (auto current_if = if_list_head; current_if != nullptr; current_if = current_if->ifa_next) {
```

Comment:
I think is whole `for` loop is broken, as it doesn't stop, if an interface was found.  Might depend on the strategy, but especially for `auto` it might not be very deterministic, which interface gets chosen.

Reply:
I will have a look at this `for` loop :+1:

Reply:
So a `break` after the interface was found was commented out. I added that again.